connect/proxy/listener.go

```diff
@@ -35,6 +35,8 @@ type Listener struct {
 
 	stopFlag int32
 	stopChan chan struct{}
+	// closed when the listener is stopped and the local socket is released.
+	stoppedChan chan struct{}
```

Comment:
Maybe also put a comment on `stopChan` - definitely a point of possible confusion as to the difference.